File: 2-UTN_PROGRA2_LABO2/0-POO_12_TRABAJOS_HERCULES/EJ1_PERSONA_MOD_ACCESO/Persona.cpp
#include "Persona.h"

int Persona::getDocumento(){
  return _documento;
}

void Persona::setDocumento(int documento){
  _documento = documento;
}

std::string Persona::getNombre(){
  return _nombre;
}

void Persona::setNombre(std::string nombre){
  _nombre = nombre;
}

std::string Persona::getApellido(){
  return _apellido;
}

void Persona::setApellido(std::string apellido){
  _apellido = apellido;
}

int Persona::getNacimiento(){
  return _nacimiento;
}

void Persona::setNacimiento(int nacimiento){
  _nacimiento = nacimiento; 
}

Persona::Persona(int documento, std::string nombre, std::string apellido, int nacimiento){
  _documento = documento;
  _nombre = nombre;
  _apellido = apellido;
  _nacimiento = nacimiento;
}

std::string Persona::nombreCompleto(){
  std::string nombreApellido;

  nombreApellido = _nombre + " " + _apellido;

  return nombreApellido;
}

int Persona::edad(){
  return 2024 - _nacimiento;
}

void Persona::tuNuevoNombre(std::string nuevoNombre){
  if(nuevoNombre != " ")//si esta vacio no lo carga
    _nombre = nuevoNombre;
}


std::string Persona::toString(){
  return "DNI " + std::to_string(_documento) + " " + nombreCompleto() + " nacio en " +  std::to_string (_nacimiento) + " -> tiene: " + std::to_string(edad()) + " años"; 
}

File: 4)LEG_AD_MS2-PROG4/4-Prog4/X-Archivos/A.1)Tp1Plan_Viejo/TP3_GRUPO_12/src/Ejercicio1/Persona.java
package Ejercicio1;

public class Persona implements Comparable<Persona> {

	private String nombre;
    private String apellido;
    private String dni;

	public Persona(){
		nombre = "John";
		apellido = "Doe";
		dni = "0";
	}

	public Persona(String nombre, String apellido, String dni) {
		this.nombre = nombre;
		this.apellido = apellido;
		this.dni = dni;
	}


	public String getNombre() {return nombre;}
	public void setNombre(String nombre) {this.nombre = nombre;}

	public String getApellido() {return apellido;}
	public void setApellido(String apellido) {this.apellido = apellido;}

	public String getDni() {return dni;}
	public void setDni(String dni) {this.dni = dni;}

	@Override
    public int compareTo(Persona persona) {
        return this.apellido.compareTo(persona.apellido);
    }

	@Override
    public String toString() {
		return nombre + " - " + apellido + " - " + dni;
    }

}
